SimCalorimetry/PhaseIAnalyzer/plugins/PhaseIAnalyzer.cc

```diff
@@ -0,0 +1,254 @@
+
+  Char_t histo[200];
+
+  sprintf(histo, "EcalDigiTaskBarrelOccupancy");
+  meEBDigiOccupancy_ = fs->make<TH2D>(histo, histo, 360, 0., 360., 170, -85., 85.);
+
+  sprintf(histo, "EcalDigiTaskBarrelOccupancyHigh");
+  meEBDigiOccupancyHigh_ = fs->make<TH2D>(histo, histo, 360, 0., 360., 170, -85., 85.);
+
+  sprintf(histo, "EcalDigiTaskBarrelOccupancyLow");
+  meEBDigiOccupancyLow_ = fs->make<TH2D>(histo, histo, 360, 0., 360., 170, -85., 85.);
+
+  sprintf(histo, "EcalDigiTaskBarrelOccupancyMid");
+  meEBDigiOccupancyMid_ = fs->make<TH2D>(histo, histo, 360, 0., 360., 170, -85., 85.);
+
+  sprintf(histo, "EcalDigiTaskBarrelDigisMultiplicity");
+  meEBDigiMultiplicity_ = fs->make<TH1D>(histo, histo, 612, 0., 61200);
+
+  for (unsigned int i = 0; i < ecalPh1::sampleSize; i++) {
+    sprintf(histo, "EcalDigiTaskBarrelAnalogPulse%02d", i + 1);
+    meEBDigiADCAnalog_[i] = fs->make<TH1D>(histo, histo, 4000, 0., 400.);
+
+    sprintf(histo, "EcalDigiTaskBarrelADCPulse%02dGain0_Saturated", i + 1);
+    meEBDigiADCgS_[i] = fs->make<TH1D>(histo, histo, 4096, -0.5, 4095.5);
+    sprintf(histo, "EcalDigiTaskBarrelGainPulse%02d", i + 1);
+    meEBDigiGain_[i] = fs->make<TH1D>(histo, histo, 4, 0, 4);
+  }
+
+  sprintf(histo, "EcalDigiTaskBarrelPedestalForPreSample");
+  meEBPedestal_ = fs->make<TH1D>(histo, histo, 4096, -0.5, 4095.5);
+
+  sprintf(histo, "EcalDigiTaskBarrelMaximumPositionGt100ADC");
+  meEBMaximumgt100ADC_ = fs->make<TH1D>(histo, histo, 10, 0., 10.);
+
+  sprintf(histo, "EcalDigiTaskBarrelMaximumPositionGt10ADC");
+  meEBMaximumgt10ADC_ = fs->make<TH1D>(histo, histo, 10, 0., 10.);
+
+  //   sprintf (histo, "EcalDigiTask Barrel ADC counts after gain switch" ) ;
+  //meEBnADCafterSwitch_ = fs->make<TH1D>histo, histo, 10, 0., 10.) ;
+}
+
+PhaseIAnalyzer::~PhaseIAnalyzer() {
+  // do anything here that needs to be done at desctruction time
+  // (e.g. close files, deallocate resources etc.)
+}
+
+//
+// member functions
+//
+
+// ------------ method called for each event  ------------
+void PhaseIAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  Handle<EBDigiCollection> pDigiEB;
+  iEvent.getByToken(digiTokenEB_, pDigiEB);
+
+  std::vector<double> ebAnalogSignal;
+  std::vector<double> ebADCCounts;
+  std::vector<double> ebADCGains_temp;
+  std::vector<double> ebADCGains;
+
+  ebAnalogSignal.reserve(EBDataFrame::MAXSAMPLES);
+  ebADCCounts.reserve(EBDataFrame::MAXSAMPLES);
+  ebADCGains_temp.reserve(EBDataFrame::MAXSAMPLES);
+  ebADCGains.reserve(EBDataFrame::MAXSAMPLES);
+
+  int nDigis = 0;
+  int LowGain(0), MidGain(0), HighGain(0);
+  int maxADCValue = 0;
+
+  for (EBDigiCollection::const_iterator pDigi = pDigiEB->begin(); pDigi != pDigiEB->end(); ++pDigi) {
+    LowGain = 0;
+    MidGain = 0;
+    HighGain = 0;
+    maxADCValue = 0;
+
+    EBDataFrame digi(*pDigi);
+    int nrSamples = digi.size();
+    std::cout << "NSamples found: " << nrSamples << std::endl;
```

Comment:
cout is not allowed in production code
(but is allowed in a standalone test analyzer, see suggestion above to move the code)